mayastor/src/lvs/lvol.rs

```diff
@@ -259,6 +275,13 @@ impl Lvol {
         let blob = unsafe { self.0.as_ref().blob };
         assert_ne!(blob.is_null(), true);
 
+        if self.is_snapshot() {
+            warn!("ignoring set property on snapshot {}", self.name());
+            return Ok(());
+        }
+        if self.is_read_only() {
+            warn!("{} is read-only", self.name());
```

Comment:
we fall through here, is that intentional?

Reply:
Sort of. The blob can become read-only for various reasons, such as if it becomes corrupt, somehow.